python/pyarrow/tests/test_convert_pandas.py

```diff
@@ -1404,6 +1404,57 @@ def test_empty_list_roundtrip(self):
 
         tm.assert_frame_equal(result, df)
 
+    def test_empty_list_metadata(self):
+        # Create table with array of empty lists, forced to have type
+        # list(string) in pyarrow
+        c1 = [["test"], ["a", "b"], None]
+        c2 = [[], [], []]
+        arrays = {
```

Comment:
You will need to use `OrderedDict` here probably to get the same ordering of the results in all Python versions.

Reply:
Yep, thanks! I saw that on the CI :)